src/Controls/src/Core/Handlers/Items/Android/MauiRecyclerView.cs

```diff
@@ -631,5 +635,33 @@ void RemoveScrollListener()
 			ClearOnScrollListeners();
 			RecyclerViewScrollListener = null;
 		}
+
+		void UpdateEmptyViewHeaderAndFooterState()
+		{
+			if (ItemsView is StructuredItemsView structuredItemsView)
+			{
+				bool headerOrFooterChanged = false;
+				if (_emptyViewAdapter.Header != structuredItemsView.Header || _emptyViewAdapter.HeaderTemplate != structuredItemsView.HeaderTemplate)
+				{
+					_emptyViewAdapter.Header = structuredItemsView.Header;
```

Comment:
This logic is repeated with header and footer. Can simplify it creating method like:

```
bool UpdateViewAdapter(ref object currentProperty, object newProperty, ref DataTemplate currentTemplate, DataTemplate newTemplate)
{
    bool propertyChanged = false;

    if (currentProperty != newProperty || currentTemplate != newTemplate)
    {
        currentProperty = newProperty;
        currentTemplate = newTemplate;
        propertyChanged = true;
    }

    return propertyChanged;
}
```

Then use it like:

```
headerOrFooterChanged |= UpdateAdapterProperties(
            ref _emptyViewAdapter.Header,
            structuredItemsView.Header,
            ref _emptyViewAdapter.HeaderTemplate,
            structuredItemsView.HeaderTemplate);
```

And same with footer.